deepchem/feat/graph_features.py

```diff
@@ -700,7 +708,31 @@ class for a full description of the names of the properties
     return np.array(values)
 
   def _featurize(self, mol):
-    """Encodes mol as a ConvMol object."""
+    """Encodes mol as a ConvMol object.
+    If per_atom_fragmentation is True,
+    then for each molecule a list of ConvMolObjects
+    will be created"""
+
+    def per_atom(n, a):
+      """
+      Enumerates fragments resulting from mol object,
+      s.t. each fragment = mol with single atom removed (all possible removals are enumerated)
+      n - list of nodes, a - adjacency list
+      """
+      for i in range(n.shape[0]):
```

Comment:
Can you add more documentation on what's happening in this function? A triple nested for-loop is hard to understand for maintainability. Is there a way to clean up this function so it's easier to follow the logic?

Reply:
i'll try to cleanup